src/java.base/share/classes/sun/security/ssl/CipherSuite.java

```diff
@@ -972,17 +968,11 @@ static String nameOf(int id) {
             }
 
             boolean found = false;
-            for (CipherSuite cs : CipherSuite.values()) {
-                if (!cs.supportedProtocols.isEmpty()) {
-                    if (cs.name.equals(name) ||
-                            cs.aliases.contains(name)) {
-                        cipherSuites.add(cs);
-                        found = true;
-                        break;
-                    }
-                } else {
-                    // values() is ordered, remaining cipher suites are
-                    // not supported.
+            for (CipherSuite cs : allowedCipherSuites) {
```

Comment:
We can avoid this loop; look up the cipher by name first (`cs = nameOf(name)`), then check for supported protocols.
Other than that, LGTM.

Reply:
Thank you for the feedback. I have updated the loop accordingly.